.github/workflows/build-rust-cache.yml

```diff
@@ -0,0 +1,53 @@
+name: Build Rust cache
```

Comment:
This workflow should be part of the `test.yml` big workflow.

Best before the `rust_prepare` step, so it can directly use is cache.

Reply:
I've removed this from this PR for now, still have to figure out how to make it work, the `rust_lint` workflow caches way more than this